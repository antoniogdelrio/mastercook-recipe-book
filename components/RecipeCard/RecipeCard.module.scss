.card {
    display: flex;
    align-items: center;
    padding: 2rem;
    border-radius: 5px;
    gap: 8px;
    width: 100%;
    max-width: 350px;
    box-shadow: 0px 10px 15px -3px rgba(0,0,0,0.1);
    background-color: #fff;
    color: #000;
    transition: transform 0.2s;
    transform: scale(1);

    &:hover {
        transform: scale(1.05);
    }
}

.info {
    display: flex;
    flex-direction: column;
    gap: 8px;

    h3 {
        font-weight: 600;
        margin-bottom: 8px;
    }

    b {
        font-weight: 900;
    }
}
