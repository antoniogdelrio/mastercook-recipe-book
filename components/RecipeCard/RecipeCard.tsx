import Image from "next/image";
import Link from "next/link";
import styles from "./RecipeCard.module.css"

type Difficulty = "Hard" | "Medium" | "Easy"

export interface RecipeCardSchema {
    title: string,
    image: string,
    time: number,
    difficulty: string
}

export default function RecipeCard (props: RecipeCardSchema) {
    return (
        <Link
            style={{ textDecoration: 'none' }}
            href="/recipes/4"
        >
            <div className={styles.card}>
                <Image
                    src={props.image}
                    alt={props.title}
                    width={100}
                    height={100}
                    blurDataURL="data:image/jpeg;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mOcOHnyfwAGLAK4yvlhMQAAAABJRU5ErkJggg=="
                    placeholder="blur"
                />
                <div className={styles.info}>
                    <h3>{props.title}</h3>
                    <p><b>Difficulty:</b> {props.difficulty}</p>
                    <p><b>Time:</b> {props.time} minutes</p>
                </div>
            </div>
        </Link>
    )
}