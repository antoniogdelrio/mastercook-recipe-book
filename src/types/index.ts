export interface Ingredient {
    description: string,
    quantity: number,
    unit: string
}

export interface Recipe {
    id: number,
    title: string,
    image: string,
    time: number,
    difficulty: string,
    ingredients: Ingredient[],
    steps: string[],
    serve: number
}

export interface RecipeSummary extends Omit<Recipe, 'ingredients' | 'steps' | 'serve'> {}
