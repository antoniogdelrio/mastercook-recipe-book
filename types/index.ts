export interface Ingredient {
    description: string,
    quantity: number,
    unit: string
}

export interface Recipe {
    id: number,
    title: string,
    image: string,
    time: number,
    difficulty: string,
    ingredients: Ingredient[],
    steps: string[]
}

export interface RecipeSummary extends Omit<Recipe, 'ingredients' | 'steps'> {}
