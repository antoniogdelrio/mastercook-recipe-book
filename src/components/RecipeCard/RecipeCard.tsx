import Image from "next/image";
import Link from "next/link";
import { BLUR_DATA_URL } from "../../constants/general";
import { RecipeSummary } from "../../types";
import styles from "./RecipeCard.module.scss"

type Difficulty = "Hard" | "Medium" | "Easy"

export default function RecipeCard (props: RecipeSummary) {
    return (
        <Link
            style={{ textDecoration: 'none' }}
            href={`/recipes/${props.id}`}
            className={styles.card}
        >
            <Image
                src={props.image}
                alt={props.title}
                width={100}
                height={100}
                blurDataURL={BLUR_DATA_URL}
                placeholder="blur"
            />
            <div className={styles.info}>
                <h3>{props.title}</h3>
                <p><b>Difficulty:</b> {props.difficulty}</p>
                <p><b>Time:</b> {props.time} minutes</p>
            </div>
        </Link>
    )
}