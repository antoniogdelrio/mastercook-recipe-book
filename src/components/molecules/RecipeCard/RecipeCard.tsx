import Image from "next/image";
import Link from "next/link";
import { BLUR_DATA_URL } from "../../../constants/general";
import { RecipeSummary } from "../../../types";
import Typography from "../../atoms/Typography/Typography";
import styles from "./RecipeCard.module.scss"

type Difficulty = "Hard" | "Medium" | "Easy"

export default function RecipeCard (props: RecipeSummary) {
    return (
        <Link
            style={{ textDecoration: 'none' }}
            href={`/recipes/${props.id}`}
            className={styles.card}
        >
            <Image
                src={props.image}
                alt={props.title}
                width={100}
                height={100}
                blurDataURL={BLUR_DATA_URL}
                placeholder="blur"
            />
            <div className={styles.info}>
                <Typography value={props.title} type="h3" customClasses={styles['info__title']} />
                <div className={styles['info__row']}>
                    <Typography value="Difficulty:" isBold={true} />&nbsp;
                    <Typography value={props.difficulty} />
                </div>
                <div className={styles['info__row']}>
                    <Typography value="Time:" isBold={true} />&nbsp;
                    <Typography value={`${props.time} minutes`} />
                </div>
            </div>
        </Link>
    )
}