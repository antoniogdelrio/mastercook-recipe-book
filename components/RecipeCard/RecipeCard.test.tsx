import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import RecipeCard from './RecipeCard'
import { Recipe } from '../../types'

describe('<RecipeCard />', () => {
    const data : Recipe = {
        id: 1,
        title: "Chicken",
        difficulty: "Hard",
        image: "/image.jpg",
        ingredients: [],
        steps: [],
        time: 120,
    }

    it('should render the recipe card successfully', () => {
        render(<RecipeCard
            id={data.id}
            title={data.title}
            difficulty={data.difficulty}
            image={data.image}
            time={data.time}
        />)

        screen.getByRole('heading', {
          name: /chicken/i
        })
        screen.getByText(/hard/i)
        screen.getByText(/120/i)
    })

    it('should match snapshot', () => {
        const { container } = render(<RecipeCard
            id={data.id}
            title={data.title}
            difficulty={data.difficulty}
            image={data.image}
            time={data.time}
        />)

        expect(container).toMatchSnapshot()
    })
})