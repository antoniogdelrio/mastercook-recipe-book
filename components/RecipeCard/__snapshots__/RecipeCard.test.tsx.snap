// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeCard /> should match snapshot 1`] = `
<div>
  <a
    class="card"
    href="/recipes/1"
    style="text-decoration: none;"
  >
    <img
      alt="Chicken"
      data-nimg="1"
      decoding="async"
      height="100"
      loading="lazy"
      src="/_next/image?url=%2Fimage.jpg&w=256&q=75"
      srcset="/_next/image?url=%2Fimage.jpg&w=128&q=75 1x, /_next/image?url=%2Fimage.jpg&w=256&q=75 2x"
      style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
      width="100"
    />
    <div
      class="info"
    >
      <h3>
        Chicken
      </h3>
      <p>
        <b>
          Difficulty:
        </b>
         
        Hard
      </p>
      <p>
        <b>
          Time:
        </b>
         
        120
         minutes
      </p>
    </div>
  </a>
</div>
`;
